### PR TITLE
Fiber pool

### LINKED ISSUE #4: The Basics
- [x] ImGui needs to be rendered
- [x] The invoker has to be tailored for Rdr
- [x] New natives file with Rdr2 natives (most likely from [Native Reference](https://rdr.gta.menu/natives) )
- [x] Game details changed from gta to Rdr
- [ ] Fiber pool and script system or some other technique that can substitute

These are a few of the fundamental assets that need to be supplied to get this thing off the ground.

### PR DESCRIPTION
**Pull request by maybegreat48**
Closes #4

### CODE REVIEW

File: src/game/rdr/Scripts.hpp [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace rage
+{
+	class scrThread;
+}
+
+namespace YimMenu::Scripts
+{
+	rage::scrThread* FindScriptThread(joaat_t hash);
+	void RunAsScript(rage::scrThread* thread, std::function<void()> callback);
```
Comment (Yimura) [inline src/game/rdr/Scripts.hpp:12]: Might need to be declared as `extern` if you're going to include `Scripts.hpp` into multiple different files.

File: src/game/frontend/menu/Menu.cpp [hunk 2]
```diff
@@ -28,6 +30,24 @@ namespace YimMenu
 				LOG(INFO) << coords.x << "x\t" << coords.y << "y\t" << coords.z << "z";
 			}
 
+			if (ImGui::Button("Spawn Ped"))
+			{
+				LOG(INFO) << "Before spawn";
+				FiberPool::Push([] {
+					auto model_hash = "U_F_M_RHDNudeWoman_01"_J; // most models don't work
```
Comment (Yimura) [inline src/game/frontend/menu/Menu.cpp:37]: Possible because this ped doesn't have any clothes or other stuff on them?

Comment (DayibBaba) [reply src/game/frontend/menu/Menu.cpp:37]: Testing in SP, 5 different models spawned with no issues. Even MP models.

**Comment by DayibBaba** [comment]
Did you eliminate the need for `get_current()` for script specific methods?

File: src/game/frontend/menu/Menu.cpp (hunk 2 above)
Comment (DayibBaba) [reply src/game/frontend/menu/Menu.cpp:37]: Post the models that you have issues spawning

**Review by Yimura — APPROVED** [review]
LGTM

**Pull request merged**